geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala

```diff
@@ -164,13 +167,55 @@ class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with L
    *
    * Our delegated AccumuloInputFormat creates a split for each range - because we set a lot of ranges in
    * geomesa, that creates too many mappers. Instead, we try to group the ranges by tservers. We use the
-   * number of shards in the schema as a proxy for number of tservers.
+   * location assignment of the tablets to tservers to determine the number of splits returned.
    */
   override def getSplits(context: JobContext): java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    logger.debug(s"Got ${accumuloSplits.length} splits")
-    accumuloSplits
+    // Get the appropriate number of mapper splits using the following priority
+    // 1. Get splits from AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS (geomesa.mapreduce.splits.max)
+    // 2. Get splits from #tserver locations * AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER (geomesa.mapreduce.splits.tserver.max)
+    // 3. Get splits from AccumuloInputFormat.getSplits(context)
+    val grpSplitsMax: Option[Int] =
+      try {
+        val maxSplits = AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS.get.toInt
+        if (maxSplits > 0) Some(maxSplits) else None
+      } catch {
+        case e: java.lang.NumberFormatException =>
+          AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS.option.foreach(value =>
+            logger.warn(s"Unable to parse geomesa.mapreduce.splits.max = $value is not a valid Int."))
+          None
+      }
+
+    lazy val grpSplitsPerTServer: Option[Int] =
+      if (AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER.get == null) None
+      else {
+        val numLocations = accumuloSplits.flatMap(_.getLocations).distinct.head.length
```

Comment:
is the `distinct` in the right place? maybe we need another flatten in there? otherwise seems like .head.length is not actually checking all the locations